testng-core/src/main/resources/testng-1.0.xsd

```diff
@@ -0,0 +1,524 @@
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:attribute>
+            <xsd:attribute name="annotations" type="xsd:string">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        If "javadoc", TestNG will look for JavaDoc annotations in your sources,
+                        otherwise it will use JDK5 annotations.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:attribute>
+            <xsd:attribute name="time-out" type="xsd:string">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        The time to wait in milliseconds before aborting the method (if parallel="methods"),
+                        or the test (parallel="tests")
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:attribute>
+            <xsd:attribute name="skipfailedinvocationcounts" type="xsd:boolean" default="false">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        Whether to skip failed invocations.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:attribute>
+            <xsd:attribute name="data-provider-thread-count" type="xsd:string">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        An integer giving the size of the thread pool to use for parallel data providers.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:attribute>
+            <xsd:attribute name="object-factory" type="xsd:string">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        A class that implements IObjectFactory that will be used to instantiate the test objects.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:attribute>
+            <xsd:attribute name="group-by-instances" type="xsd:boolean" default="false"/>
+            <xsd:attribute name="preserve-order" type="xsd:boolean" default="true"/>
+            <xsd:attribute name="allow-return-values" type="xsd:boolean" default="false">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        If true, tests that return a value will be run as well
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:attribute>
+        </xsd:complexType>
+    </xsd:element>
+    <xsd:simpleType name="parallel">
+        <xsd:restriction base="xsd:string">
+            <xsd:enumeration value="false">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        Deprecated
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="true">
```

Comment:
Haha! Good catch. It is my objective too.

We can remove deprecated values if we upgrade the version.

At the same time, it will be easier to keep dtd for 1.0 and have xsd for 1.1

I tried to generate dto from xsd but I had gradle issues with the plugins I tried.

I plan to add it into a dedicated module.

Reply:
Values removed.